### DISCUSSION TITLE
Does MIR works on Windows

### DISCUSSION THREAD

**Discussion by rishavs** [vnmakarov/mir#345] [Q&A]
Hi

I am using Windows x86_64 as my dev device. I am able to generate a simple mir file.
How would I run this mir file as a JIT executable?

Does MIR supports Windows as of yet?

**Comment by vnmakarov** [comment]
Windows port is not finished yet.  So it is not officially supported and tested minimally.  The major obstacle is implementation of C setjmp/longjmp.

Windows port implements Win ABI.  I'd recommend to start from .github/workflows/AMD64-Linux-OSX-Windows-test.yml.  It is a github action file which is used to build an test MIR for Windows (see windows part of the file).

**Reply by rishavs** [reply]
Hi @vnmakarov , I fear I have some fundamental questions

My understanding of the overall process is

1. Wrap the libmid.dll/.so library in my bootstrapping language
2. Generate MIR binary/text file using the libmir apis
3. load the MIR code back into the libmir.dll for optimization. MIR would now JIT compile the loaded IR
4. Obtain the entry point for the compiled JIT code using the libmir api callback
5. Invoke the entry point to execute the compiled code using the libmir apis
6. What then? How would I generate the executable from this now? Is my compiler itself the final executable for my toy language program? Would that mean every single program in my toy language should come with the libmir.dll attached to it?

The final step is where I am struggling with, mainly because I do not understand the operational aspects of  JIT compiler. Can you please give me some pointers here?

One of the goals of my language is that I should be able to create executables for each program.
